packages/calypso-e2e/src/test/email-client.test.ts

```diff
@@ -0,0 +1,90 @@
+import { describe, expect, test, jest } from '@jest/globals';
+import { EmailClient } from '../email-client';
+import { SecretsManager, Secrets } from '../secrets';
+
+const fakeSecrets = {
+	testAccounts: {
+		basicUser: {
+			username: 'wpcomuser2',
+			password: 'hunter2',
+			primarySite: 'wpcomuser.wordpress.com/',
+		},
+	},
+	mailosaur: {
+		apiKey: 'fake_key',
+		defaultUserInboxId: 'fake_id',
+	},
+} as unknown as Secrets;
+
+jest.spyOn( SecretsManager, 'secrets', 'get' ).mockImplementation( () => fakeSecrets );
+
+describe( 'EmailClient: get2FACodeFromMessage', function () {
+	const baseSMSMessage = {
+		id: 'fake_id',
+		server: 'fake_server',
+		from: [
+			{
+				phone: '18888888888',
+			},
+		],
+		to: [
+			{
+				phone: '18887776666',
+			},
+		],
+		received: new Date( Date.now() - 3600000 ),
+		subject: 'SMS',
+		text: {
+			codes: [
+				{
+					value: '0123456',
+				},
+			],
+			body: 'WordPress.com verification code: 0123456',
+		},
+	};
+
+	test( 'One code in message', function () {
+		const emailClient = new EmailClient();
+
+		const code = emailClient.get2FACodeFromMessage( baseSMSMessage );
+		expect( code ).toBe( '0123456' );
+	} );
+
+	test( 'Two codes in message', function () {
+		const message = JSON.parse( JSON.stringify( baseSMSMessage ) );
```

Comment:
100% optional style suggestions!

But I run into this problem when unit testing all the time! 😆 I'll have a base object that you want to tweak throughout tests for different cases, but I need a fresh object reference each time. You can definitely do the stringify and re-parse trick, but my personal preference is to just change `baseSMSMessage` to a function --> `createBaseSMSMessage()` -- then you get a fresh reference every time without having to do any JSON tricks!

Again, totally optional as it's just a style preference thing, but thought I'd share in case it's helpful :)

Reply:
Addressed in 2f93fdf836bdf423be0769629391e14c14208962.